### PR TITLE
Westy moretests

### PR DESCRIPTION
**Pull request by westyUMass**
Additional changes for MOB-1084 (Make all single tests DefaultGNSTest so that we can package-test them)

All tests now run without failures except the ones that have been moved into the failingtests directory.
A bit more testing to do before the issue is resolved.

### CODE REVIEW

File: test/edu/umass/cs/gnsclient/client/singletests/AclAllFieldsSuperuser.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void test_100_LookupMasterGuid() {
   @Test
   public void test_142_ACLCreateAnotherGuid() {
     try {
-      String barneyName = "barney" + RandomString.randomString(6);
+      String barneyName = "barney" + RandomString.randomString(12);
```
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/singletests/AclAllFieldsSuperuser.java]: Minor: Good to have a single variable that controls the length of random string.

File: test/edu/umass/cs/gnsclient/client/singletests/ClientLNSTest.java [hunk 2]
```diff
@@ -50,7 +52,9 @@ public ClientLNSTest() {
     if (clientCommands == null) {
       try {
         clientCommands = new GNSClientCommands();
-        clientCommands.setGNSProxy(new InetSocketAddress("127.0.0.1", 24598));
+        //PaxosConfig.getActives() works here because the server and client use the same properties file.
+        InetAddress lnsAddress = PaxosConfig.getActives().values().iterator().next().getAddress();
+        clientCommands.setGNSProxy(new InetSocketAddress(lnsAddress, 24598));
```
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/singletests/ClientLNSTest.java]: Is port 24598 never going to change? 
Minor: Good to have a common variable for this port in case it changes.

File: test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java [hunk 3]
```diff
@@ -25,21 +27,21 @@
...
+
 
   /**
    *
    */
   public SimpleHttpClientTest() {
-    if (client == null) {
-      client = new HttpClient("127.0.0.1", 8080);
+    if (httpClient == null) {
+      httpClient = new HttpClient("127.0.0.1", 24703);
```
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java]: General question, not a code review comment; Why was this changed?
Since many applications rely on port 8080, I guess a documentation update is necessary so that nothing breaks. The CASA Android developers rely on port 8080, for example.

**Comment by akarthik10** [comment]
I'm ready to merge this if you think these minor comments can be incorporated at a later time.

File: test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java (hunk 3 above)
Comment (westyUMass) [inline test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java]: Arun changed this. He didn't like using 8080. I on the other hand thought it was OK :-)

File: test/edu/umass/cs/gnsclient/client/singletests/ClientLNSTest.java (hunk 2 above)
Comment (westyUMass) [inline test/edu/umass/cs/gnsclient/client/singletests/ClientLNSTest.java]: Agreed. Also need a way to lookup up the HTTP port from the tests.

**Comment by westyUMass** [comment]
Cool. Thanks. The last two are related (ports) so I'll make a Jira issue for them if I can't figure it out immediately.

File: test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java (hunk 3 above)
Comment (westyUMass) [inline test/edu/umass/cs/gnsclient/client/singletests/SimpleHttpClientTest.java]: BTW: The number is an offset from the port number of the server which is often 24403. See HTTP_PORT_OFFSET (default 300) and HTTP_PORT_SSL_OFFSET (default 400) in ReconfigurationConfig.
